ffcx/codegeneration/C/integrals_template.py

```diff
@@ -30,9 +30,55 @@
   {tabulate_tensor_float64}
   {tabulate_tensor_complex64}
   {tabulate_tensor_complex128}
+  {tabulate_tensor_cuda}
   .needs_facet_permutations = {needs_facet_permutations},
   .coordinate_element_hash = {coordinate_element_hash},
 }};
 
 // End of code for integral {factory_name}
 """
+
+cuda_wrapper = """
+
+// Begin CUDA wrapper for integral {factory_name}
+void tabulate_tensor_cuda_{factory_name}(int* num_program_headers,
+                                         const char*** program_headers,
+                                         const char*** program_include_names,
+                                         const char** out_program_src,
+                                         const char** tabulate_tensor_function_name)
+{{
+  const char* program_src = ""
+    "#define alignas(x)\\n"
+    "#define restrict __restrict__\\n"
+    "\\n"
+    "typedef unsigned char uint8_t;\\n"
```

Comment:
I saw Stackoverflow thread on this https://stackoverflow.com/questions/4189243/whats-a-good-alternative-to-uint8-t-when-it-is-not-provided-by-the-compiler; is it not possible to use `#include <stdint.h>`? These typedefs are likely to cause issues so it would be good to avoid them.

If it's unavoidable then we need a comment here with an explanation.

Reply:
Unfortunately it looks like this is unavoidable. I've added a comment explaining the use of typedefs. Another relevant thread is https://stackoverflow.com/questions/50565200/including-c-standard-headers-in-cuda-nvrtc-code.